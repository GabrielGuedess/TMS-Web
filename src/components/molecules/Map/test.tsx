import { render } from '@testing-library/react';

import { Map } from '.';

jest.mock('mapbox-gl', () => ({
  Marker: jest.fn(() => ({
    setLngLat: jest.fn(() => ({
      addTo: jest.fn(),
    })),
  })),
  Map: jest.fn(() => ({
    on: jest.fn(),
    addLayer: jest.fn(),
    otherMethods: jest.fn(),
    getStyle: () => ({ layers: {} }),
  })),
}));

describe('<Map />', () => {
  it('should render the map correctly', () => {
    const { container } = render(
      <Map
        center={[-46.527_312_8, -23.467_588_2]}
        marker={[-46.527_312_8, -23.467_588_2]}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
