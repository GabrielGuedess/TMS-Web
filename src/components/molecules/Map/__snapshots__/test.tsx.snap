// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should render the map correctly 1`] = `
<div
  class="h-[40rem] w-full rounded-sm"
  id="map"
/>
`;
